dashboard/test/models/concerns/pd/form_test.rb

```diff
@@ -32,6 +62,16 @@ class Pd::FormTest < ActiveSupport::TestCase
     assert_equal ["can't be blank"], form.errors.messages[:form_data]
   end
 
+  test 'pd form only validates form data when changed' do
+    form = DummyFormWithRequiredFields.new
+    form.form_data = {firstField: "foo"}.to_json
+    form.save
+
+    assert form.valid?
+    form.form_data = {firstField: "bar"}.to_json
+    assert_equal false, form.valid?
+  end
+
```

Comment:
ahh, yeah, I can see how this is unclear.

That is essentially what I'm doing here; the only difference is that this dummy form I've constructed doesn't actually perform validation on save. Or, in other words, it _always_ treats a save call as a `save(validate: false)`.

Do you think adding comments explaining exactly what's going on here would be sufficiently clear? Or do I need to rethink this whole "DummyForm" idea?